x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_investigation/graph_investigation.tsx

```diff
@@ -211,6 +222,28 @@ export const GraphInvestigation = memo<GraphInvestigationProps>(
       },
     });
 
+    const nodeDetailsClickHandler = useCallback(
+      (node: NodeProps) => {
+        onOpenEventPreview?.(node.data);
+      },
+      [onOpenEventPreview]
+    );
+
+    const { nodeExpandPopover, labelExpandPopover, openPopoverCallback } = useGraphPopovers({
+      dataViewId: dataView?.id ?? '',
+      searchFilters,
+      setSearchFilters,
+      nodeDetailsClickHandler: onOpenEventPreview ? nodeDetailsClickHandler : undefined,
```

Comment:
when `onOpenEventPreview` is not available we won't render `show event details` option